datafusion-examples/examples/rewrite_expr.rs

```diff
@@ -59,7 +59,7 @@ pub fn main() -> Result<()> {
 
     // then run the optimizer with our custom rule
     let optimizer = Optimizer::with_rules(vec![Arc::new(MyOptimizerRule {})]);
-    let optimized_plan = optimizer.optimize(&analyzed_plan, &config, observe)?;
+    let optimized_plan = optimizer.optimize(analyzed_plan, &config, observe)?;
```

Comment:
This illustrates the API change -- the optimizer now takes an owned plan rather than a reference

Reply:
A great progress!